src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -136,7 +136,16 @@ private void Initialize()
                 // machine connecting to public server).
 
                 // SMTP RFC's require ASCII only host names in the HELO/EHLO message.
-                string clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
+                string clientDomainRaw;
+                try
+                {
+                    clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
+                }
+                catch (NotSupportedException)
```

Comment:
Added the comment. (Though it kinda seemed like a reasonable safeguard either way. But I don't feel strongly one way or the other.)